Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/duration/TraceLocationWarnDurationFragment.kt

```diff
@@ -0,0 +1,168 @@
+    private val viewModel: TraceLocationWarnDurationViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TraceLocationWarnDurationViewModel.Factory
+            factory.create(navArgs.traceLocation)
+        }
+    )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        enterTransition = MaterialSharedAxis(MaterialSharedAxis.Z, true)
+        returnTransition = MaterialSharedAxis(MaterialSharedAxis.Z, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.state.observe2(this) { uiState ->
+            with(binding) {
+                if (uiState.description != null) {
+                    description.text = uiState.description
+                }
+
+                if (uiState.startDateTime != null && uiState.endDateTime != null) {
+
+                    val startTime = uiState.startDateTime!!.toDateTime()
+                    val endTime = uiState.endDateTime!!.toDateTime()
+
+                    eventDate.isGone = false
+
+                    val startDay = startTime.toLocalDate().toString("dd.MM.yyyy")
+                    val startHour = startTime.toLocalTime().toString("HH:mm")
+                    val endDay = endTime.toLocalDate().toString("dd.MM.yyyy")
+                    val endHour = endTime.toLocalTime().toString("HH:mm")
+                    eventDate.text = if (startTime.toLocalDate() == endTime.toLocalDate()) {
+                        requireContext().getString(
+                            R.string.trace_location_organizer_detail_item_duration,
+                            startDay,
+                            startHour,
+                            endHour
+                        )
+                    } else {
+                        requireContext().getString(
+                            R.string.trace_location_organizer_detail_item_duration_multiple_days,
+                            startDay,
+                            startHour,
+                            endDay,
+                            endHour
+                        )
+                    }
+                } else {
+                    eventDate.isGone = true
+                }
+
+                startInputEdit.setText(uiState.formattedDateTime())
+                durationInputEdit.setText(uiState.duration.toContactDiaryFormat())
+            }
+        }
+
+        with(binding) {
+            startInputEdit.setOnClickListener {
+                it.hideKeyboard()
+                showDatePicker()
+            }
+            durationInputEdit.setOnClickListener {
+                it.hideKeyboard()
+                showDurationPicker()
+            }
+            toolbar.setNavigationOnClickListener {
+                popBackStack()
+            }
```

Comment:
what about the `Next` button ? I think we need to create Parcelable and pass the Tracelocation and the dates to the TAN screen. or do you want to it in the next ticket?